pkg/ruler/evaluator_remote.go

```diff
@@ -341,6 +344,18 @@ func metricToLabels(m model.Metric) labels.Labels {
 	return b.Labels()
 }
 
+func detailToQueryTags(detail rules.RuleDetail) []string {
```

Comment:
This really feels like a responsibility the ruler should not have.
In the `httpreq` we should have a helper which allows additional values to be injected into and extracted from these headers.

Reply:
I don't disagree. The query tags need some work and a standard format (seems logfmt would do). That said this blows up the work for this small enhancement.